cmd/bucket-handlers.go

```diff
@@ -326,6 +316,15 @@ func (api objectAPIHandlers) DeleteMultipleObjectsHandler(w http.ResponseWriter,
 
 	var dErrs = make([]error, len(deleteObjects.Objects))
 	for index, object := range deleteObjects.Objects {
+		var s3Error APIErrorCode
+		if s3Error = checkRequestAuthType(ctx, r, policy.DeleteObjectAction, bucket, object.ObjectName); s3Error != ErrNone {
+			// In the event access is denied, a 200 response should still be returned
+			// http://docs.aws.amazon.com/AmazonS3/latest/API/multiobjectdeleteapi.html
+			if s3Error != ErrAccessDenied {
+				writeErrorResponse(ctx, w, errorCodes.ToAPIErr(s3Error), r.URL, guessIsBrowserReq(r))
+				return
+			}
+		}
```

Comment:
note that for multi-objects delete, if there were partial success we need to communicate it back to the client appropriately.

Reply:
Yes. In that case should we collect all the errors and if we see any error other than `AccessDenied`, return this error?

Reply:
i think we don't need to differentiate between access-denied and other errors (apart from object-not-found)
you can check how aws-s3 behaves and we can behave accordingly

Reply:
done